crates/file_finder/src/file_finder_tests.rs

```diff
@@ -1541,6 +1541,107 @@ async fn test_search_results_refreshed_on_adding_and_removing_worktrees(
     });
 }
 
+#[gpui::test]
+async fn test_selected_match_stays_selected_after_matches_refreshed(cx: &mut gpui::TestAppContext) {
+    let app_state = init_test(cx);
+
+    app_state.fs.as_fake().insert_tree("/src", json!({})).await;
+
+    app_state
+        .fs
+        .create_dir("/src/even".as_ref())
+        .await
+        .expect("unable to create dir");
+
+    let initial_files_num = 5;
+    for i in 0..initial_files_num {
+        let filename = format!("/src/even/file_{}.txt", 10 + i);
+        app_state
+            .fs
+            .create_file(Path::new(&filename), Default::default())
+            .await
+            .expect("unable to create file");
+    }
+
+    let project = Project::test(app_state.fs.clone(), ["/src".as_ref()], cx).await;
+    let (workspace, cx) = cx.add_window_view(|cx| Workspace::test_new(project.clone(), cx));
+
+    // Initial state
+    let picker = open_file_picker(&workspace, cx);
+    cx.simulate_input("file");
+    let selected_index = 3;
+    // Checking only the filename, not the whole path
+    let selected_file = format!("file_{}.txt", 10 + selected_index);
+    // Select even/file_13.txt
+    for _ in 0..selected_index {
+        cx.dispatch_action(SelectNext);
+    }
+
+    picker.update(cx, |finder, _| {
+        assert_match_selection(finder, selected_index, &selected_file)
+    });
+
+    // Add more matches to the search results
+    let files_to_add = 10;
+    for i in 0..files_to_add {
+        let filename = format!("/src/file_{}.txt", 20 + i);
+        app_state
+            .fs
+            .create_file(Path::new(&filename), Default::default())
+            .await
+            .expect("unable to create file");
+    }
+    cx.executor().advance_clock(FS_WATCH_LATENCY);
+
+    // file_13.txt is still selected
```

Comment:
Can we ensure that something had changed after entries were added? 
Because, if nothing was added then of course the selection did not change too.

Reply:
I believe, there's an implicit check in the way how we assert that the same entry is selected, since the expected index is increased by the number of added entries. If nothing has changed, then the index should stay the same as well, thus failing the assert.
https://github.com/zed-industries/zed/blob/4c5208059b31e5493843cafcd0f5e3ddb21d4f20/crates/file_finder/src/file_finder_tests.rs#L1598

Reply:
Sure, we can use that, never mind me then.